_scripts/git-precommit-hook.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# 1. Get the list of all modified and dependent packages.
+# Cleans up yarn's output by removing everything before the package name, the first and the last line.
+PACKAGES_MODIFIED=`yarn workspaces list --since=HEAD^ -R | cut -d/ -f2 | sed '1d' | sed '$d'`
+
+# 2. If no packages were modified, return early; otherwise we'd build all packages.
+if [[ $PACKAGES_MODIFIED == "" ]]; then
+    echo "no modified packages found"
+    exit 0
+fi
+
+echo "modified packages found"
+
+# 3. Build all back-end TS packages in the list
+ORIGINAL_IFS=$IFS
+IFS=$'\n'
+# We don't need to worry about front-end packages that use TS, since they must build successfully for CI to pass.
+BACKEND_PACKAGES=( "fxa-auth-server" "fxa-event-broker" "fxa-graphql-api" "fxa-shared" "fxa-support-panel" "fxa-admin-server" )
```

Comment:
nit: Not sure if this list is in a special order, or if we should A-z it so "fxa-admin-server" is first so it's alphabetical.
Or, if they _are_ in a special build order, maybe a bonus comment/warning not to reorder them, lest we anger the build gods.

:+1: Also verified that each of these packages has a `compile` script (which we knew from the diffs below, but was mostly curious if other packages had the same command name as well):

```sh
git grep -n '"compile": ' packages/*/package.json | cat

packages/fxa-admin-server/package.json:8:    "compile": "tsc --noEmit",
packages/fxa-auth-server/package.json:13:    "compile": "tsc --noEmit",
packages/fxa-event-broker/package.json:9:    "compile": "tsc --noEmit",
packages/fxa-graphql-api/package.json:8:     "compile": "tsc --noEmit",
packages/fxa-shared/package.json:13:         "compile": "tsc --noEmit",
packages/fxa-support-panel/package.json:11:  "compile": "tsc --noEmit",
```

Reply:
The order of the array doesn't matter, since `yarn workspaces foreach ... run compile` will order the processing based on the dependency graph. But I can make it alphabetical just to be a little easier to read/modify in the future. Thanks for the suggestion.